heidrun/ia.rb

```diff
@@ -0,0 +1,238 @@
+    #   META.xml <creator>
+    creator class: DPLA::MAP::Agent,
+            each: record.field('creator'),
+            as: :creator do
+      providedLabel creator
+    end
+
+    # dc:date
+    #   META.xml <date>
+    date class: DPLA::MAP::TimeSpan,
+         each: record.field('date'),
+         as: :date do
+      providedLabel date
+    end
+
+    # dcterms:description
+    #   meta.xml <description>
+    description record.field('description')
+
+    # dcterms:extent
+    #   MARC.XML <datafield tag="300">[any subfields]
+    extent record.field('marc', 'record', 'datafield')
+      .match_attribute(:tag, '300')
+
+    # dcterms:identifier
+    #   META.XML
+    #     <identifier> may be the same thing;
+    #     <call_number>Call number: "[value]"</call_number>;
+    #     <datafield tag="035" ind1=" " ind2=" ">
+    #       <subfield code="a">[value]</subfield>
+    identifier record.map(&identifier_map).flatten
+
+    # dcterms:language
+    #   META.XML <language>
+    language record.field('language')
+
+    # dcterms:spatial
+    #   MARC.xml
+    #     <datafield tag="6##"><subfield code="z">
+    #     [ANY 600-699 data field
+    #      but only subfield "z" which is a geographic indicator]
+    spatial class: DPLA::MAP::Place,
+            each: record.field('marc', 'record', 'datafield')
+                .match_attribute(:tag) { |tag| tag.start_with?('6') }
+                .field('subfield')
+                .match_attribute(:code, 'z'),
+            as: :place do
+      providedLabel place
+    end
+
+    # dcterms:publisher
+    #   meta.xml <publisher>
+    publisher class: DPLA::MAP::Agent,
+              each: record.field('publisher'),
+              as: :publisher do
+      providedLabel publisher
+    end
+
+    # dc:relation
+    #   MARC.XML
+    #     <datafield tag="440"> or
+    #     <datafield tag="490"> or
+    #     <datafield tag="800"> or
+    #     <datafield tag="810"> or
+    #     <datafield tag="830">;
+    #     <datafield tag="785">
+    #       [Add ". " between subfields; Do not include subfield w];
+    #     <datafield tag="780">
+    #       [Add ". " between subfields; Do not include subfield w]
+    relation record.map(&relation_map).flatten
+
+    # dc:rights
+    #   "Access to the Internet Archive's Collections is granted
+    #    for scholarship and research purposes only. Some of the
+    #    content available through the Archive may be governed by
+    #    local, national, and/or international laws and regulations,
+    #    and your use of such content is solely at your own risk."
+    rights 'Access to the Internet Archive’s Collections is granted
+            for scholarship and research purposes only. Some of the
+            content available through the Archive may be governed by
+            local, national, and/or international laws and regulations,
+            and your use of such content is solely at your own risk.'
+                  .gsub(/\s+/, ' ')
```

Comment:
> I notice the UTF-8 apostrophe character in "Archive’s" and have some hard-to-place intuition that it might cause some issues down the line as part of a string that's most likely to get copied and pasted or read into spreadsheets, etc. Would it be better off as originally specified with the boring old ASCII apostrophe?

:+1:, from my perspective. My guess this is from Google Docs probably autocorrecting the apostrophe...